app/src/main/java/org/sopt/certi/presentation/ui/personalInfo/component/PersonalInfoNicknameTextField.kt

```diff
@@ -60,15 +63,24 @@ fun PersonalInfoNicknameTextField(
             )
             PersonalInfoTextFieldButton(
                 text = "중복확인",
-                onClick = onButtonClick,
-                isEnable = value != preValue
+                onClick = {
+                    onButtonClick()
+                    focusManager.clearFocus()
+                },
+                isEnable = isEnable
             )
         }
         BasicTextField(
             value = value,
-            onValueChange = {
-                if (it.length <= 7) {
-                    onValueChange(it)
+            onValueChange = { newValue ->
+                if (newValue == value) {
+                    return@BasicTextField
+                }
+                if (newValue.contains("\n") || newValue.contains("\r")) {
+                    return@BasicTextField
+                }
+                if (newValue.length <= 7) {
+                    onValueChange(newValue)
                 }
```

Comment:
[2]: 여기 부분이 어떤 로직인건지 궁금합니디

Reply:
singleLine 옵션을 써도 엔터 키 입력 시 \n이 포함되어 넘어오는 경우가 있어 이를 막기 위해 추가했더요
글구 한국어가 자음과 모음을 조합하는 언어라... 엔터 누르면 글자 확정 이벤트가 발생하는데 이런 경우 텍스트 내용은 같아도 onValueChange가 호출되면서 중복 확인 상태가 초기화되는 이슈가 있어서 🥺 기존 값이랑 동일한 경우도 onValueChange 실행 안 되도록 햇서요

근데 만약 복붙 가능한 상황도 고려하면
```suggestion
            onValueChange = { newValue ->
                val filteredValue = newValue.replace("\n", "").replace("\r", "")
                if (filteredValue == value) return@BasicTextField
                onValueChange(filteredValue.take(7))
            },
```
이게 나을것같긴 하네여